src/Microsoft.OData.Core/JsonLight/ODataJsonLightContextUriParser.cs

```diff
@@ -61,29 +61,35 @@ private ODataJsonLightContextUriParser(IEdmModel model, Uri contextUriFromPayloa
         /// <param name="model">The model to use when resolving the target of the URI.</param>
         /// <param name="contextUriFromPayload">The string value of the odata.metadata annotation read from the payload.</param>
         /// <param name="payloadKind">The payload kind we expect the context URI to conform to.</param>
-        /// <param name="clientCustomTypeResolver">The function of client cuetom type resolver.</param>
+        /// <param name="clientCustomTypeResolver">The function of client custom type resolver.</param>
         /// <param name="needParseFragment">Whether the fragment after $metadata should be parsed, if set to false, only MetadataDocumentUri is parsed.</param>
         /// <param name="throwIfMetadataConflict">Whether to throw if a type specified in the ContextUri is not found in metadata.</param>
+        /// <param name="baseUri">Optional value (with default value of null) of base Uri used for resolving the context url.
+        /// It should be a non-null value when resolving a top-level relative context Uri.

```

Comment:
>a top-level  [](start = 57, length = 12)

Why call out "a top level"? Shouldn't it be non-null for resolving *any* relative context Url?  Note that relative nested context urls should be resolved using the parent absolute contextUrl as a base; is this supported?

Reply:
Should be any relative context url. Removed "a top level". 

---
In reply to: [203898403](https://github.com/OData/odata.net/pull/1211#discussion_r203898403) [](ancestors = 203898403)

Reply:
But changing the comment doesn't change how this works.
If I have a contextUrl nested within another contextUrl "scope", that nested context URL needs to be resolved relative to the outer contextUrl.

For example:

~~~json
{
   "@context" : "[http://sampleservice/$externalMetadata#employees](http://sampleservice/$externalMetadata#employees)",
   "value" : [
     {
       "id":"00001",
       "name":"Kathy Smith",
       ...
       "company" :
          {
              "@context": "#company",
              "name":"FirstCompany"
          }
     }
   ]
}
~~~
or
~~~json
{
   "@context" : "[http://sampleservice/externalMetadata#company](http://sampleservice/externalMetadata#company)",
   "name":"FirstCompany",
   "employees@context": "#employees"
   "value" : [
     {
       "id":"00001",
       "name":"Kathy Smith",
       ...
     }
   ]
}

It's possible (likely) that our stack doesn't support this type of nested contextUrls, but I still want to make sure that we can support relative urls if we add support for nested context Urls in the future.

---
In reply to: [204168642](https://github.com/OData/odata.net/pull/1211#discussion_r204168642) [](ancestors = 204168642,203898403)

Reply:
Since we are not sure how exactly nested contextUrl is going to be supported, its relative url support should be part of the future task when it comes up. Right?

---
In reply to: [205245558](https://github.com/OData/odata.net/pull/1211#discussion_r205245558) [](ancestors = 205245558,204168642,203898403)